lib/sign_in/widgets/form.dart

```diff
@@ -82,48 +67,62 @@ class _SignInFormState extends State<SignInForm> {
           ),
-          SizedBox(height: 16),
-          SignInInput(
-            controller: _emailController,
-            label: s.signIn_form_email,
-            error: _isEmailValid ? null : s.signIn_form_email_error,
-            onChanged: () => setState(() {}),
-          ),
-          SizedBox(height: 16),
-          SignInInput(
-            controller: _passwordController,
-            label: s.signIn_form_password,
-            obscureText: true,
-            error: _isPasswordValid ? null : s.signIn_form_password_error,
-            onChanged: () => setState(() {}),
-          ),
-          SizedBox(height: 16),
-          MorphingLoadingButton(
-            onPressed: _signIn,
-            isLoading: _isLoading,
-            child: Padding(
-              padding: EdgeInsets.all(12),
-              child: Text(
-                _isLoading ? s.general_loading : s.signIn_form_signIn,
-                style: TextStyle(color: Colors.white, fontSize: 20),
-              ),
+          SizedBox(height: 32),
+          ConstrainedBox(
+            constraints: BoxConstraints(
+              maxHeight: mediaQuery.size.height -
+                  400 -
+                  mediaQuery.padding.bottom -
+                  mediaQuery.padding.top,
             ),
+            child: WebView(
+                javascriptMode: JavascriptMode.unrestricted,
+                onWebViewCreated: (controller) => this.controller = controller,
+                onPageFinished: (url) async {
+                  if (url ==
+                      'https://${services.get<AppConfig>().domain}/login') {
+                    // The JavaScript is meant to isolate the login section
+                    // Hopefully there will be an option to only get that
+                    // in a non-hacky way in the future.
+                    await controller.evaluateJavascript('''
+                  var node = document.getElementById('loginarea');
+                  var html = document.getElementsByTagName('html')[0];
+                  html.removeChild(html.childNodes[2]);
+                  html.appendChild(node);
+                  document.getElementsByTagName('h2')[0].innerHTML = ''
+                  ''');
+                  } else if (url ==
+                      'https://${services.get<AppConfig>().domain}/dashboard') {
+                    var cookies =
+                        await controller.evaluateJavascript('document.cookie');
+                    // Yes, this is not elegant. You may complain about it
+                    // when there is a nice way to get a single cookie via JavaScript.
+                    var jwt = cookies
+                        .split('; ')
+                        .firstWhere((element) => element.startsWith('"jwt='))
+                        .replaceAll('"', '')
+                        .substring(4);
+
+                    final storage = services.get<StorageService>();
+                    await storage.token.setValue(jwt);
```

Comment:
Just storing the new JWT isn't enough. Have a look at `StorageService.setUserInfo` which also sets `userId` and `email`—we need to at least store `userId`, maybe the email address can be retrieved via the API

Reply:
i think there is a /me route

Reply:
We can also retrieve it from the token like we did some time ago: https://github.com/schul-cloud/schulcloud-flutter/blob/3be9922ea19286fa6e4d855f8bb3f9208e190146/lib/app/services/storage.dart#L20-L28